### PR TITLE
[docs] Improve some phrasing in the installation guide

### PR DESCRIPTION
**Pull request by Lewington-pitsos**
I tried to improve the phrasing of the guide while keeping the actual semantics exactly the same.

@StrikerRUS perhaps you could review this since you seem to be active in this area?

### CODE REVIEW

File: docs/Installation-Guide.rst [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
 
-Here is the guide for the build of LightGBM CLI version.
+This is a guide for building various versions of the LightGBM Command Line Interface (CLI). If you want to build the Python-package or R-package please refer to `Python-package`_ and `R-package`_ folders respectively.
 
-All instructions below are aimed to compile 64-bit version of LightGBM.
-It is worth to compile 32-bit version only in very rare special cases of environmental limitations.
-32-bit version is slow and untested, so use it on your own risk and don't forget to adjust some commands in this guide.
+All instructions below are aimed at compiling the 64-bit version of LightGBM.
+It is worth compiling the 32-bit version only in very rare special cases involving environmental limitations.
+The 32-bit version is slow and untested, so use it on your own risk and don't forget to adjust some of the commands below when installing.
```
Comment (jameslamb) [inline docs/Installation-Guide.rst:8]:
```suggestion
The 32-bit version is slow and untested, so use it at your own risk and don't forget to adjust some of the commands below when installing.
```

File: docs/Installation-Guide.rst [hunk 2]
```diff
@@ -1,21 +1,19 @@
 Installation Guide
 ==================
 
-Here is the guide for the build of LightGBM CLI version.
+This is a guide for building various versions of the LightGBM Command Line Interface (CLI). If you want to build the Python-package or R-package please refer to `Python-package`_ and `R-package`_ folders respectively.
```
Comment (jameslamb) [inline docs/Installation-Guide.rst:4]:
```suggestion
This is a guide for building the LightGBM Command Line Interface (CLI). If you want to build the Python-package or R-package please refer to `Python-package`_ and `R-package`_ folders respectively.
```

I think "various versions" here is a little too wordy

File: docs/Installation-Guide.rst [hunk 3]
```diff
@@ -191,12 +189,12 @@ Build Threadless Version (not Recommended)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 The default build version of LightGBM is based on OpenMP.
-However, you can build the LightGBM without OpenMP support, but it is **strongly not recommended**.
+You can build the LightGBM without OpenMP support but it is **strongly not recommended**.
```
Comment (jameslamb) [inline docs/Installation-Guide.rst:192]:
```suggestion
You can build LightGBM without OpenMP support but it is **strongly not recommended**.
```

I know this was there before, but while you're touching this line we might as well clean up this "the LightGBM"

**Comment by StrikerRUS** [comment]
@Lewington-pitsos 
> I tried to improve the phrasing of the guide while keeping the actual semantics exactly the same.

Thanks a lot for making docs clearer! I believe it is really important.

> perhaps you could review this since you seem to be active in this area?

I'm sorry, I'm not a native English speaker, so I suppose @jameslamb will be really better reviewer for these changes (many thanks for the prompt review @jameslamb !). But I'll take a look at this tomorrow anyway.

**Comment by Lewington-pitsos** [comment]
Thanks @jameslamb for the additions, they have been added.

Glad you think the commits are useful @StrikerRUS, I really liked the guide (it's great to see a properly in-depth installation manual) so I thought I'd see if i could help out.
